### PR TITLE
Updated architecture page

### PR DESCRIPTION
**Pull request by JohanMabille**
- Made a new scheme for Projects ovcerview section
- Moved the Projects overview section to the top of the page
- Added xeus to the kernel section

### CODE REVIEW

File: docs/source/projects/architecture/content-architecture.rst
Comment (choldgraf) [inline docs/source/projects/architecture/content-architecture.rst:26]: Non-blocking note here: this is why I like using "one line per sentence". Avoids arbitrary re-line-breaking changes like this :-)

File: docs/source/projects/architecture/content-architecture.rst
Comment (choldgraf) [inline docs/source/projects/architecture/content-architecture.rst:15]: How is this image being generated? Do we have docs about it somewhere so that somebody else could make edits? Could we provide a URL link to instructions as a comment or something?

File: docs/source/projects/architecture/content-architecture.rst
Comment (choldgraf) [inline docs/source/projects/architecture/content-architecture.rst:80]: I'd suggest turning this into a three-bullet list so that each is easier to notice. e.g.:

```
- **Wrapper kernels** are ...
- **Native kernels** implement ...
- **xeus-based kernels** implement ...
```

File: docs/source/projects/architecture/content-architecture.rst
Comment (choldgraf) [inline docs/source/projects/architecture/content-architecture.rst:92]:
```suggestion
to write when the language interpreter provides a C++ or a C API.
```

File: docs/source/projects/architecture/content-architecture.rst
Comment (choldgraf) [reply docs/source/projects/architecture/content-architecture.rst:15]: Not in this PR, but I'd suggest that in the future we use [excalidraw](https://excalidraw.com/) for these kinds of drawings. This would let us export the drawing as an SVG with the ExcaliDraw metadata embedded inside it, so a maintainer could make edits by loading the SVG back into excalidraw, adding it back here, and then making a PR to update the SVG.

I'll open an issue to track this, just noting it here.

**Review by choldgraf — APPROVED** [review]
This looks good to me - I have a few quick comments in there but none of them are blocking!

File: docs/source/projects/architecture/content-architecture.rst
Comment (JohanMabille) [reply docs/source/projects/architecture/content-architecture.rst:15]: This image is generated with [drawio](https://app.diagrams.net/), the scheme is embedded in both images (png and svg), so anyone can edit it.

Comment (choldgraf) [reply docs/source/projects/architecture/content-architecture.rst:15]: Is that documented somewhere? Maybe I missed it, anyway as I say it doesn't need to block the PR

**Comment by JohanMabille** [comment]
Thanks for your review @choldgraf !

File: docs/source/projects/architecture/content-architecture.rst
Comment (choldgraf) [inline docs/source/projects/architecture/content-architecture.rst:14]:
```suggestion

.. To edit this file, import it into diagrams.net:  https://app.diagrams.net/
```

File: docs/source/projects/architecture/content-architecture.rst
Comment (choldgraf) [inline docs/source/projects/architecture/content-architecture.rst:116]:
```suggestion

.. To edit this file, import it into diagrams.net:  https://app.diagrams.net/
```

File: docs/source/projects/architecture/content-architecture.rst
Comment (choldgraf) [inline docs/source/projects/architecture/content-architecture.rst:78]:
```suggestion

.. To edit this file, import it into diagrams.net:  https://app.diagrams.net/
```

**Review by choldgraf — APPROVED** [review]
I'll add a comment about diagrams.net to each `image` directive so others know how to edit it - will commit those and merge away if doc build is happy. This is a nice improvement, it makes me happy any time somebody improves the documentation :-)

**Pull request merged**
